### PR TITLE
Add group list command

### PR DESCRIPTION
**Pull request by aaschaer**
Adds groups service to login and group  list command to use it

One thing we might want to confirm is the fields, I'm calling `enforce_session` "High Assurance" but maybe that's incorrect. And we could potentially do some parsing of the role sub-documents to give information there

### CODE REVIEW

File: src/globus_cli/commands/group/list.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    groups = client.get_my_groups()
+
+    formatted_print(
+        groups,
+        fields=[
+            ("Group ID", "id"),
+            ("Name", "name"),
+            ("Type", "group_type"),
+            ("High Assurance", "enforce_session"),
```
Comment (sirosen) [inline src/globus_cli/commands/group/list.py:21]: Although I would think `High Assurance` is a fine/good name in isolation, after looking at the webapp I think we should change this to `Session Enforcement` with values of `strict` and `not strict`. I think that's just
```python
lambda x: "strict" if x["enforce_session"] else "not strict"
```

Here's how the webapp presents this info to a group admin:
image


While I'd love for us to be consistent across services and call this `High Assurance` _everywhere_, we can't be consistent both with the webapp and with the GCP/GCS context. Given that choice, I think that using a different name for this in the CLI vs webapp will create more confusion than having a different name between Groups and Collections.

File: tests/functional/test_group.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    data = load_api_fixtures("groups.yaml")
+
+    group1_id = data["metadata"]["group1_id"]
+    group2_id = data["metadata"]["group2_id"]
+    group1_name = data["metadata"]["group1_name"]
+    group2_name = data["metadata"]["group2_name"]
+
+    result = run_line("globus group list")
+
+    assert group1_id in result.output
```
Comment (sirosen) [inline tests/functional/test_group.py:14]: I wonder... Would it be useful to have a regex matching helper for tests like this?
Something like
```python
# some "matcher fixture"
assert matcher(
    r"^Group ID:\s+(.*)$", # a regex with capturing groups
    result.output, # input string
    groups=[group1_id]  # values compared against the captured groups, in order
)
```

**Review by sirosen — COMMENTED** [review]
> we could potentially do some parsing of the role sub-documents to give information there

I think we should do this.

Specifically, I think a field `("Roles", _comma_delimited_roles_names)` would do the trick.
Just the `"role"` field is sufficient. I don't think the average user cares very much which of their identities is associated with a specific group.

File: tests/functional/test_group.py (hunk 2 above)
Comment (aaschaer) [reply tests/functional/test_group.py:14]: Hmmm, yeah that might be a good idea. I mostly looked at the output from running the command to make sure it looked good, but if something changed and messed it up this test likely wouldn't catch it

Comment (sirosen) [reply tests/functional/test_group.py:14]: I'll queue it up for later. It should be pretty easy to write something.

**Review by sirosen — APPROVED** [review]

**Pull request merged**